concordium-smart-contract-testing/src/lib.rs

```diff
@@ -0,0 +1,2745 @@
+        &mut self,
+        sender: AccountAddress,
+        module_path: P,
+    ) -> Result<SuccessfulModuleDeployment, DeployModuleError> {
+        // Load file
+        let module = std::fs::read(module_path)?;
+        self.module_deploy_aux(sender, &module)
+    }
+
+    /// Deploy a wasm module as it is output from `cargo concordium build`, i.e.
+    /// **including** the prefix of 4 version bytes and 4 module length bytes.
+    pub fn module_deploy<P: AsRef<Path>>(
+        &mut self,
+        sender: AccountAddress,
+        module_path: P,
+    ) -> Result<SuccessfulModuleDeployment, DeployModuleError> {
+        // Load file
+        let module = std::fs::read(module_path)?;
+        // Here, we skip the 8 bytes that encode the smart contract version and module
+        // length
+        self.module_deploy_aux(sender, &module[8..])
+    }
+
+    pub fn contract_init(
+        &mut self,
+        sender: AccountAddress,
+        module_reference: ModuleReference,
+        contract_name: ContractName,
+        parameter: ContractParameter,
+        amount: Amount,
+        energy_reserved: Energy,
+    ) -> Result<SuccessfulContractInit, ContractInitError> {
+        // Lookup artifact
+        let artifact = self.get_artifact(module_reference)?;
+        let mut transaction_fee = self.calculate_energy_cost(self.lookup_module_cost(&artifact));
+        // Get the account and check that it has sufficient balance to pay for the
+        // reserved_energy and amount.
+        let account_info = self.get_account(sender)?;
+        if account_info.balance < self.calculate_energy_cost(energy_reserved) + amount {
+            return Err(ContractInitError::InsufficientFunds);
+        }
+        // Construct the context.
+        let init_ctx = v0::InitContext {
+            metadata:        ChainMetadata {
+                slot_time: self.slot_time,
+            },
+            init_origin:     sender,
+            sender_policies: account_info.policies.clone(),
+        };
+        // Initialize contract
+        let mut loader = v1::trie::Loader::new(&[][..]);
+        let res = v1::invoke_init(
+            artifact,
+            init_ctx,
+            v1::InitInvocation {
+                amount,
+                init_name: contract_name.get_chain_name(),
+                parameter: &parameter.0,
+                energy: InterpreterEnergy {
+                    // TODO: Why do we have two separate energy types?
+                    energy: energy_reserved.energy,
+                },
+            },
+            false,
+            loader,
+        )
+        .map_err(|e| ContractInitError::StringError(e.to_string()))?;
+        // Handle the result and update the transaction fee.
+        let res = match res {
+            v1::InitResult::Success {
+                logs,
+                return_value: _, /* Ignore return value for now, since our tools do not support
+                                  * it for inits, currently. */
```

Comment:
If you write a smart contract with concordium_std then there won't be a return value on inits, so it would be confusing that you can see a return value when testing. At least that was my initial thought. I don't have very strong opinions about it.